lib/ftp/src/ftp.erl

```diff
@@ -2087,9 +2090,9 @@ handle_caller(#state{caller = {transfer_data, {Cmd, Bin, RemoteFile}}} =
 %% Connect to FTP server at Host (default is TCP port 21)
 %% in order to establish a control connection.
 -spec setup_ctrl_connection(Host, Port, Timeout, State) -> Result when
-      Host    :: inet:socket_address() | inet:hostname(),
+      Host    :: inet:ip_address() | inet:hostname(),
       Port    :: inet:port_number(),
-      Timeout :: timeout(),
+      Timeout :: integer(),
```

Comment:
Is this right?

`connect/5` supports `infinity` timeout, however this does not because of the case below. Was the intention to support `infinity` and implementation is wrong, or there was no expectation of supporting this.

Reply:
AFAIK, it is supposed to be an `non_neg_integer()`. 

From the usage of the record, the documentation, and the [code documentation](https://github.com/erlang/otp/pull/6545/files/9407e3d3402c28aa2d1ad3065a5f65384ffabae8#diff-bbc31375ed1562b0f69299e0b126ac627bfd216c1164f0c36a38d46cafb1c0f8L86), it is a `non_neg_integer()`.

It is probably not too hard to support `infinity`, which AFAIK, means that there is no timeout to consider :)
If someone using the FTP app really needs this, we can look into it :)

Reply:
Makes sense. I this case I think it does make sense to keep it just as an integer

Reply:
why `integer()` instead of `non_neg_integer()`? we know that a timeout must be a positive number, correct?
Just wondering :)

Reply:
Yeah. The more precise type makes sense here. As far as eqwalizer is concerned though, it doesn't distinguish various numeric types, see https://github.com/WhatsApp/eqwalizer/blob/main/docs/reference/types.md#numeric-types

Reply:
For traceability, I created a ticket [eqWA-11](https://github.com/WhatsApp/eqwalizer/issues/11) :)
I hope this is ok